src/utils/multicaller.ts

```diff
@@ -0,0 +1,45 @@
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+import set from 'lodash.set';
+import { multicall } from '../utils';
+
+export default class Multicaller {
```

Comment:
@CjHare Great work on PR! but multicall and Multicaller here causes duplicate code from snapshot.js, we should find some way to make them usable from one place only ( maybe try [patching snapshot.js](https://www.npmjs.com/package/patch-package) )

Reply:
@bonustrack any ideas? or these can be here? 🤔

Reply:
@ChaituVR you're quite right, the only way I could see avoiding the duplication would be to patch `snapshot.js` in some way. 
From a design perspective I can appreciate `snapshot.js` not needing such a change (as it's a change for the purpose of testing) ...but I can always start that conversation 🙂